src/addons/dispatch/Header.svelte

```diff
@@ -1,50 +1,134 @@
 <script lang="ts">
+  import Button from "../../common/Button.svelte";
+  import BackArrow from "../../common/icons/BackArrow.svelte";
+  import ShareIcon from 'svelte-octicons/lib/Share16.svelte';
+  import GitHubIcon from "svelte-octicons/lib/MarkGithub16.svelte";
   import type { AddOnListItem } from "../browser/AddOnListItem.svelte";
   import { _ } from "svelte-i18n";
+  import Hashtag from "../../common/icons/Hashtag.svelte";
 
   export let addon: AddOnListItem;
 
   $: author = addon.author || addon.repository.split("/")[0];
+
+  async function onShare() {
+    // TODO Provide user feedback upon sharing success or error.
+    // Right now using alert() which is okay but not ideal.
+    try {
+      if (navigator.share) {
+        await navigator.share({ url: window.location.href, title: document.title });
+      } else if (navigator.clipboard) {
+        await navigator.clipboard.writeText(window.location.href);
+        alert($_("addonDispatchDialog.shareClipboardSuccess"));
+      }
+    } catch (error) {
+      alert(error.message);
+    }
+  }
```

Comment:
@eyeseast Here is an instance where a lightweight feedback mechanism for users would be useful. For now, relying on browser's built-in `alert` to provide feedback, but since it takes over the page it's not ideal.